### PR TITLE
[TikTok] Extract auto captions

### LINKED ISSUE #2184: [TikTok] Add subtitles support
### Checklist

- [X] I'm reporting a site feature request
- [X] I've verified that I'm running yt-dlp version **2021.12.27**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Example URLs

https://www.tiktok.com/@hankgreen1/video/7047596209028074758

### Description

TikTok has built-in captioning support which is used widely since you can automatically generate captions when posting. They only show up in the mobile apps (not the web version) and the creator can choose the vertical position at which they appear.

image

Supporting these in yt-dlp would be great for accessibility.

### Verbose log

```shell
[debug] Command-line config: ['-Uv', '--list-subs', 'https://www.tiktok.com/@hankgreen1/video/7047596209028074758']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.9.6 (CPython 64bit) - Linux-5.13.12-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2021.12.27, Current version: 2021.12.27
yt-dlp is up to date (2021.12.27)
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@hankgreen1/video/7047596209028074758
[TikTok] 7047596209028074758: Downloading video details
[debug] Sort order given by extractor: quality, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, hdr:12(7), asr, proto, vext, aext, hasaud, source, id
7047596209028074758 has no subtitles
```

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.
Closes https://github.com/yt-dlp/yt-dlp/issues/2184
Drafted because no tests yet + no text positioning info

### CODE REVIEW

**Comment by pukkandan** [comment]
this can be merged even without position info imo.

How did you implement this without being able to test?

**Comment by MinePlayersPE** [comment]
@pukkandan Okay to merge?

**Review by pukkandan — CHANGES_REQUESTED** [review]
Ive had this review written for a while, but never submitted 😓

File: yt_dlp/extractor/tiktok.py [hunk 1]
```diff
@@ -218,6 +238,7 @@ def extract_addr(addr, add_meta={}):
             'artist': music_author,
             'timestamp': int_or_none(aweme_detail.get('create_time')),
             'formats': formats,
+            'subtitles': subtitles,
```
[the remark this answers is not in the document]
Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py]: Since former is more common, i'll just use that ¯\\\_(ツ)_/¯

**Pull request merged**